### ISSUE TITLE
scipy.special.sh_jacobi weights have become wrongly scaled

### ISSUE THREAD

**Issue by gdmcbain** [scipy/scipy#5170]
w = sh_jacobi(5, 2, 2).weights
    print w[:, 0].dot(w[:, 1])

`w` should contain the abscissæ and weights of the five-point Gauss--Jacobi quadrature rule with weight `x` over the unit segment (0, 1) in its first two columns; e.g. `(w[:, 0]**k).dot(w[:, 1])` should integrate `x**k` with weight `x`, i.e. `x**(k+1)`, and therefore give `1.0/(k+2)`.  It did in `scipy.version.version` 0.14.1 but in 0.15.1 it gives `4.0/(k+2)`.  

The weights seem to have become been scaled four times too high.  This result applies for all integer `n > k / 2` (or something like that, but of course n not too big).

The code for `sh_jacobi` itself doesn't seem to have changed so I suspect the error is in one of the internal routines it calls; thus, it may well affect other orthogonal polynomial functions.

**Comment by argriffing** [comment]
This code gives 1/3 in development numpy/scipy.
```python
>>> import numpy as np
>>> import scipy
>>> np.__version__
'1.10.0.dev0+9232200'
>>> scipy.__version__
'0.17.0.dev0+db2ab40'
>>> from scipy.special import sh_jacobi
>>> w = sh_jacobi(5, 2, 2).weights
>>> w[:, 0].dot(w[:, 1])
0.33333333333333343
```

**Comment by pv** [comment]
This was fixed in 0.16.0.
https://github.com/scipy/scipy/commit/fc5f34825d07b281016fb8f29129bca4996f6403
